hazelcast/src/main/java/com/hazelcast/version/Version.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.version;
+
+import com.hazelcast.internal.cluster.impl.ClusterDataSerializerHook;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Comparator;
+
+/**
+ * Determines Hazelcast version.
+ */
+@SuppressWarnings("checkstyle:magicnumber")
+public final class Version implements IdentifiedDataSerializable, Comparable<Version>, Serializable {
+
+
```

Comment:
Jaromir had a couple of comments to this class in my PR (but I see it's a bit different now) 
Here's the link: https://github.com/hazelcast/hazelcast/pull/9229